### PR TITLE
Epic: Resource Versions

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Close: #[issue-number]

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/block/block.service.ts [hunk 1]
```diff
@@ -942,4 +942,35 @@ export class BlockService {
...
+      where: {
+        ...args.where,
+        resourceId: args.where.resourceId,
+        deletedAt: null,
+      },
+      include: {
+        versions: {
+          where: {
+            versionNumber: {
+              not: CURRENT_VERSION_NUMBER,
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/block/block.service.ts:959]:
```suggestion
              gt: CURRENT_VERSION_NUMBER,
```

File: packages/amplication-server/src/core/block/block.service.ts [hunk 2]
```diff
@@ -942,4 +942,35 @@ export class BlockService {
...
+          take: 1,
+          orderBy: {
+            versionNumber: Prisma.SortOrder.desc,
+          },
+        },
+        parentBlock: true,
+      },
+    });
+
+    return (await blocks).map((block) => {
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/block/block.service.ts:971]: Why do you need to await blocks again? you already await the findMany call above

File: packages/amplication-server/src/core/block/block.service.ts (hunk 1 above)
Comment (yuval-hazaz) [reply packages/amplication-server/src/core/block/block.service.ts:959]: CURRENT_VERSION_NUMBER is 0 - so both will give the same results.
we are using not in entities so i prefer to keep the same code
the latest version will always be larger then zero as long as there is a version - so the filter is actually "give me top 1 version from all published versions"

File: packages/amplication-server/src/core/resourceVersion/resourceVersion.service.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    });
+
+    const logger = this.logger.child({
+      resourceVersionId: resourceVersion.id,
+      resourceId: resourceVersion.resourceId,
+      userId: resourceVersion.userId,
+      user,
+    });
+
+    const resource = await this.resourceService.findOne({
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/resourceVersion/resourceVersion.service.ts:76]: Don't you want to check this before creating the version in the DB at line 51?

**Review by mulygottlieb — APPROVED** [review]

File: packages/amplication-server/src/util/version.ts [hunk 4]
```diff
@@ -1 +1 @@
-export const version = "3.3.2";
+export const version = "3.2.3";
```
Comment (yuval-hazaz) [inline packages/amplication-server/src/util/version.ts:1]:
```suggestion
export const version = "3.3.2";

```

File: packages/amplication-client/src/util/version.ts [hunk 5]
```diff
@@ -1 +1 @@
-export const version = "3.3.2";
+export const version = "3.2.3";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/util/version.ts:1]:
```suggestion
export const version = "3.3.2";

```

**Pull request merged**